webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,65 @@
+import React, { createContext } from 'react';
+import { Modal } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import Header from './subcomponents/ForemanModalHeader';
+import Footer from './subcomponents/ForemanModalFooter';
+// import { translate as __ } from '../../common/I18n';
+
+export const ModalContext = createContext(null);
+
+const ForemanModal = props => {
+  const { isOpen, onClose, title } = props;
+  if (!isOpen) return null;
+  // Extract header and footer from children, if provided
+  let children = React.Children.toArray(props.children);
+  const headerChild = children.find(
+    child => child.type === ForemanModal.Header
+  );
+  const footerChild = children.find(
+    child => child.type === ForemanModal.Footer
+  );
+  children = children.filter(
+    child => child !== headerChild && child !== footerChild
+  );
+  // Use context to provide values to all descendants (not just direct children)
+  const context = {
+    isOpen,
+    onClose,
+    title,
+  };
+  // Change the name of the props we are passing down to Modal to conform to Patternfly 3 api
+  const newProps = Object.assign({}, props);
+  delete newProps.onClose;
+  delete newProps.isOpen;
+
+  return (
+    <ModalContext.Provider value={context}>
+      <aside className="foreman-modal-content" style={{ padding: '1em' }}>
```

Comment:
why do we need to use the `aside`?
can you move the inline styling to css? thanks

Reply:
@laviro The `aside` is because I wanted to think about semantic HTML and making things reasonably accessible.  I read [this StackOverflow answer](https://stackoverflow.com/questions/3072597/semantically-accurate-html5-element-for-a-modal-dialog) and like the idea of it.